### PR TITLE
feat(console): partially implement console key down handler

### PR DESCRIPTION
**Pull request by natrist**

### CODE REVIEW

File: src/console/Handlers.cpp [hunk 1]
```diff
@@ -15,8 +16,28 @@ int32_t OnIdle(const EVENT_DATA_IDLE* data, void* param) {
...
+        ConsoleSetActive(active);
+
+        if (!active) {
+            // TODO
+            // ResetHighlight();
+        }
+
+        return 0;
+    }
+    else if (EventIsKeyDown(ConsoleGetHotKey()) || !ConsoleGetActive()) {
```
Comment (fallenoak) [inline src/console/Handlers.cpp:33]:
```suggestion
    }

    if (EventIsKeyDown(ConsoleGetHotKey()) || !ConsoleGetActive()) {
```

File: src/console/Handlers.cpp [hunk 2]
```diff
@@ -15,8 +16,28 @@ int32_t OnIdle(const EVENT_DATA_IDLE* data, void* param) {
...
+        return 1;
+    }
+    else {
+        // TODO
+
+        return 0;
+    }
```
Comment (fallenoak) [inline src/console/Handlers.cpp:40]:
```suggestion
    }

    // TODO
    return 0;
```

**Review by fallenoak — CHANGES_REQUESTED** [review]

File: src/console/Handlers.cpp (hunk 1 above)
Comment (fallenoak) [reply src/console/Handlers.cpp:33]: We can treat these early conditionals as guard conditions. In a guard condition, when the condition is met and the condition returns from the function, there's no need to use `else` to chain conditions. Each condition can just be fully separate `if` block, as I've suggested in my review.

Comment (natrist) [reply src/console/Handlers.cpp:33]: I thought you specifically asked to do that in a previous PR review. Will apply proposed changes.

Comment (fallenoak) [reply src/console/Handlers.cpp:33]: Great! Hmm, not sure offhand which review that was, but if you link it, we can talk through what might have been different in that case (or if I provided bad advice).

**Review by fallenoak — APPROVED** [review]

**Pull request merged**